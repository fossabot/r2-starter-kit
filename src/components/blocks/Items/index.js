import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import ProgressiveImage from 'components/elements/ProgressiveImage';

class Items extends PureComponent {
  static propTypes = {
    posts: PropTypes.array,
    handleClick: PropTypes.func,
  }
  static defaultProps = {
    posts: [],
    handleClick: () => {},
  }
  render() {
    const { posts, handleClick } = this.props;
    return (
      <div className="ts text container">
        <div className="ts divided items">
          {posts.map(({
            id,
            title,
            description,
            image,
            placeholder,
          }, key) => (
            <div
              key={`item${key}`}
              className="item"
              style={{ cursor: 'pointer' }}
              onClick={() => handleClick(id)}
            >
              <div className="image">
                <ProgressiveImage
                  url={image}
                  placeholder={placeholder}
                />
              </div>
              <div className="content">
                <div className="header">
                  {title}
                </div>
                <div className="description">{description}</div>
              </div>
            </div>
          ))}
        </div>
      </div>
    );
  }
}

export default Items;
