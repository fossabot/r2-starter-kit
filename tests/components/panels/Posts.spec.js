import React from 'react';
import { shallow } from 'enzyme';

import Component from 'components/panels/Posts';
import Items from 'components/blocks/Items';

describe('<Posts />', () => {
  const initProps = {
    uiState: 'idle',
    Items: <Items />,
  };

  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<Component {...initProps} />);
  });

  it('should render initial page', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with uiState prop which value is "loading"', () => {
    wrapper.setProps({ uiState: 'loading' });
    expect(wrapper).toMatchSnapshot();
  });
});
