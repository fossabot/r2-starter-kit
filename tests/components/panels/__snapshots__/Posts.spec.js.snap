// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should render component with uiState prop which value is "loading" 1`] = `
<Fragment>
  <Loading />
</Fragment>
`;

exports[`<Posts /> should render initial page 1`] = `
<Fragment>
  <Items
    handleClick={[Function]}
    posts={Array []}
  />
</Fragment>
`;
